### PR TITLE
RavenDB-15750 Take leverage from ANTLR and use it for intellisense in Studio

### PR DESCRIPTION
**Pull request by ml054**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-15750

### Additional description

Include details of the change made in this Pull Request or additional notes for the solution. Anything that can be useful for reviewers of this PR.

_Please delete below the options that are not relevant_

### Type of change
- New feature

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works
- It has been verified by manual testing

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/languageService/grammar/BaseRqlLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,662 @@
+lexer grammar BaseRqlLexer;
```
Comment (ayende) [inline src/Raven.Studio/languageService/grammar/BaseRqlLexer.g4:1]: Maybe move the syntax to a different location?

File: src/Raven.Studio/languageService/src/providers/keywords.ts [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            completions.push(AutocompleteKeywords.handleOrOperator());
+        }
+        
+        if (candidates.tokens.has(RqlParser.AND)) {
+            completions.push(AutocompleteKeywords.handleAndOperator());
+        }
+        
+        completions.push(...AutocompleteKeywords.handleRootKeywords(candidates, parser, parseTree, writtenText));
+
+        AutocompleteKeywords.debugRemainingTokens(candidates, parser, writtenText); //TODO: comment out!
```
Comment (ml054) [inline src/Raven.Studio/languageService/src/providers/keywords.ts:309]: lt's leave it for some period of time - useful for debugging

File: src/Raven.Studio/languageService/src/generated/BaseRqlLexer.interp [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+null
+null
+null
+null
+null
+null
+null
+null
+null
+null
```
Comment (ayende) [inline src/Raven.Studio/languageService/src/generated/BaseRqlLexer.interp:88]: ??

Comment (ml054) [reply src/Raven.Studio/languageService/src/generated/BaseRqlLexer.interp:88]: @ayende all files in src/generated are automatically generated by ANTLR - I commited them as it requires to have java runtime to compile. This is trade-off.

Comment (ml054) [reply src/Raven.Studio/languageService/src/generated/BaseRqlLexer.interp:88]: same for queries.json file- @maciejaszyk created this by runing all RavenDB tests - it allows us to battle test our grammar.

**Pull request merged**